src/Microsoft.Extensions.ApiDescription.Client/DownloadFile.cs

```diff
@@ -94,19 +96,116 @@ public async Task<bool> ExecuteAsync()
+                        {
+                            // Check hashes before using the downloaded information.
+                            var downloadHash = GetHash(responseStream);
+                            responseStream.Position = 0L;
+
+                            byte[] destinationHash;
+                            using (var destinationStream = File.OpenRead(destinationPath))
+                            {
+                                destinationHash = GetHash(destinationStream);
+                            }
+
+                            var sameHashes = downloadHash.Length == destinationHash.Length;
+                            for (var i = 0L; sameHashes && i < downloadHash.Length; i++)
+                            {
+                                sameHashes = downloadHash[i] == destinationHash[i];
+                            }
+
+                            if (sameHashes)
+                            {
+                                log.LogMessage($"Not overwriting existing and matching file '{destinationPath}'.");
+                                return;
+                            }
+                        }
+                        else
+                        {
+                            // May need to create directory to hold the file.
+                            var destinationDirectory = Path.GetDirectoryName(destinationPath);
+                            if (!(string.IsNullOrEmpty(destinationDirectory) || Directory.Exists(destinationDirectory)))
+                            {
+                                Directory.CreateDirectory(destinationDirectory);
+                            }
+                        }
+
+                        // Create or overwrite the destination file.
+                        reachedCopy = true;
+                        using (var outStream = File.Create(destinationPath))
+                        {
+                            responseStream.CopyTo(outStream);
+                        }
+                    }
+                }
+            }
+            catch (HttpRequestException ex) when (destinationExists)
+            {
+                if (ex.InnerException is SocketException socketException)
+                {
+                    log.LogWarning($"Unable to download {uri}, socket error code '{socketException.SocketErrorCode}'.");
+                }
+                else
+                {
+                    log.LogWarning($"Unable to download {uri}: {ex.Message}");
+                }
+            }
+            catch (Exception ex)
+            {
+                log.LogError($"Downloading '{uri}' failed.");
+                log.LogErrorFromException(ex, showStackTrace: true);
+                if (reachedCopy)
+                {
+                    File.Delete(destinationPath);
+                }
+            }
+        }
+
+        private static byte[] GetHash(Stream stream)
+        {
+            using (var algorithm = SHA256.Create())
```

Comment:
Do we need this - https://github.com/aspnet/Mvc/blob/release/2.2/src/Microsoft.AspNetCore.Mvc.Razor/Infrastructure/CryptographyAlgorithms.cs?